From d1641d016eb9dcc12f5a9941f6777d891979158b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tor=20Arne=20Vestb=C3=B8?= <tor.arne.vestbo@qt.io>
Date: Thu, 11 Sep 2025 15:15:09 +0200
Subject: [PATCH 053/553] macOS: Show icons in menus on macOS 26 and above

As this is the new system behavior.

Pick-to: 6.9 6.8 6.5
Change-Id: Ie33bc70a627133459e26411f4492ea0fe1f07ca4
Reviewed-by: Timur Pocheptsov <timur.pocheptsov@qt.io>
(cherry picked from commit 08a17345db77589fc5a1c451df2795071b5bd3e9)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/plugins/platforms/cocoa/qcocoatheme.mm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/platforms/cocoa/qcocoatheme.mm b/src/plugins/platforms/cocoa/qcocoatheme.mm
index e3ff518b4e1..12a1bc2fd00 100644
--- a/src/plugins/platforms/cocoa/qcocoatheme.mm
+++ b/src/plugins/platforms/cocoa/qcocoatheme.mm
@@ -471,7 +471,7 @@ QVariant QCocoaTheme::themeHint(ThemeHint hint) const
     case QPlatformTheme::KeyboardAutoRepeatRate:
         return 1.0 / NSEvent.keyRepeatInterval;
     case QPlatformTheme::ShowIconsInMenus:
-        return false;
+        return QOperatingSystemVersion::current() >= QOperatingSystemVersion::MacOSTahoe;
     case QPlatformTheme::MenuSelectionWraps:
         return false;
     default:
-- 
2.51.2

